frontend/src/PagesComponents/Organization_/Project_/Updates.elm

```diff
@@ -410,6 +419,22 @@ update urlLayout zone now urlInfos organizations projects msg model =
         CustomModalClose ->
             ( model |> setModal Nothing, Cmd.none )
 
+        Undo ->
+            case model.history of
+                [] ->
+                    ( model, "Can't undo, action history is empty" |> Toasts.info |> Toast |> T.send )
+
+                ( back, next ) :: history ->
+                    update "undo" urlLayout zone now urlInfos organizations projects back { model | history = history, future = ( back, next ) :: model.future }
```

Comment:
Undo is calling again `update` function with a different `doCmd` parameter and moving the last action into future actions